scripts/test_env.sh

```diff
@@ -90,24 +94,85 @@ function bulk_export {
 }
 
 function run_peer () {
-    PEER="$TEST/peers/$1"
+    PEER="$TEST/peers/iroha$1"
     mkdir -p "$PEER"
     STORAGE="$PEER/storage"
     mkdir -p "$STORAGE"
     KURA_BLOCK_STORE_PATH="$STORAGE"
-    TORII_P2P_ADDR="$HOST:${p2p_ports[$1]}"
-    TORII_API_URL="$HOST:${api_ports[$1]}"
-    TORII_TELEMETRY_URL="$HOST:${telemetry_ports[$1]}"
-    IROHA_PUBLIC_KEY=${public_keys[$1]}
-    IROHA_PRIVATE_KEY="{ \"digest_function\": \"ed25519\", \"payload\": \"${private_keys[$1]}\" }"
-    exec -a "$1" "$TEST/peers/iroha" "$2" > "$PEER/.log" & disown
+    TORII_P2P_ADDR="${hosts[$1]}:${p2p_ports[$1]}"
+    TORII_API_URL="${hosts[$1]}:${api_ports[$1]}"
+    TORII_TELEMETRY_URL="${hosts[$1]}:${telemetry_ports[$1]}"
+    IROHA_PUBLIC_KEY="${public_keys[$1]}"
+    IROHA_PRIVATE_KEY="${private_keys[$1]}"
+    exec -a "iroha$1" "$TEST/peers/iroha" "$2" > "$PEER/.log" & disown
+}
+
+function run_n_peers {
+    generate_genesis_key_pair
+    for peer in $(seq 0 $(($1-1)))
+    do
+       generate_p2p_port $peer
+       generate_api_port $peer
+       generate_telemetry_port $peer
+       generate_peer_key_pair $peer
+       hosts[$peer]="$HOST"
+    done
+    SUMERAGI_TRUSTED_PEERS="$(generate_trusted_peers $1)"
+    for peer in $(seq 1 $(($1-1)))
+    do
+        run_peer $peer
+    done
+    run_peer 0 --submit-genesis
 }
 
-function run_4_peers {
-    run_peer iroha1
-    run_peer iroha2
-    run_peer iroha3
-    run_peer iroha0 --submit-genesis
+function generate_docker_compose {
+    generate_genesis_key_pair
+    for peer in $(seq 0 $(($1-1)))
+    do
+       generate_p2p_port $peer
+       generate_api_port $peer
+       generate_telemetry_port $peer
+       generate_peer_key_pair $peer
+       hosts[$peer]="iroha$peer"
+    done
+    SUMERAGI_TRUSTED_PEERS="$(generate_trusted_peers $1)"
+
+cat <<eof
+version: "3.8"
+services:
+eof
+
+    for peer in $(seq 0 $(($1-1)))
+    do
+        if [ $peer -eq 0 ]
+        then
+            SUBMIT_GENESIS="--submit-genesis"
+        else
+            SUBMIT_GENESIS=""
+        fi
+
+cat <<eof
+  iroha$peer:
+    image: hyperledger/iroha2:dev
+    environment:
+      TORII_P2P_ADDR: '${hosts[$peer]}:${p2p_ports[$peer]}'
+      TORII_API_URL: '${hosts[$peer]}:${api_ports[$peer]}'
+      TORII_TELEMETRY_URL: '${hosts[$peer]}:${telemetry_ports[$peer]}'
+      IROHA_PUBLIC_KEY: '${public_keys[$peer]}'
+      IROHA_PRIVATE_KEY: '${private_keys[$peer]}'
+      SUMERAGI_TRUSTED_PEERS: '$SUMERAGI_TRUSTED_PEERS'
```

Comment:
Strip it out. The docker compose should be generated by a special program. We used to have `iroha_swarm` for iroha 1. 

More on this in https://github.com/hyperledger/iroha/issues/2373